src/optimagic/optimizers/pyswarms_optimizers.py

```diff
@@ -0,0 +1,904 @@
+    """Configuration for the Ring topology.
+
+    This algorithm uses a fixed ring topology where particles are connected to their
+    local neighbors. This parameter allows customization of the number of neighbors,
+    distance metric, and whether the topology remains static throughout optimization.
+
+    """
+
+    stopping_maxiter: PositiveInt = STOPPING_MAXITER
+    """Maximum number of iterations."""
+
+    initial_positions: list[PyTree] | None = None
+    """Option to set the initial particle positions.
+
+    If None, positions are generated randomly within the given bounds, or within [0, 1]
+    if bounds are not specified.
+
+    """
+
+    oh_strategy: dict[str, str] | None = None
+    """Dictionary of strategies for time-varying options."""
+
+    boundary_strategy: Literal[
+        "periodic", "reflective", "shrink", "random", "intermediate"
+    ] = "periodic"
+    """Strategy for handling out-of-bounds particles.
+
+    Available options: periodic (default),
+    reflective, shrink, random, intermediate.
+
+    """
+
+    velocity_strategy: Literal["unmodified", "adjust", "invert", "zero"] = "unmodified"
+    """Strategy for handling out-of-bounds velocities.
+
+    Available options: unmodified (default),
+    adjust, invert, zero.
+
+    """
+
+    velocity_clamp_min: float | None = None
+    """Minimum velocity limit for particles."""
+
+    velocity_clamp_max: float | None = None
+    """Maximum velocity limit for particles."""
+
+    convergence_ftol_rel: NonNegativeFloat = CONVERGENCE_FTOL_REL
+    """Stop when relative change in objective function is less than this value."""
+
+    convergence_ftol_iter: PositiveInt = 1
+    """Number of iterations to check for convergence."""
+
+    n_cores: PositiveInt = 1
+    """Number of cores for parallel evaluation."""
+
+    center_init: PositiveFloat = 1.0
+    """Scaling factor for initial particle positions."""
+
+    verbose: bool = False
+    """Enable or disable the logs and progress bar."""
+
+    seed: int | None = None
+    """Random seed for reproducibility."""
+
+    def _solve_internal_problem(
+        self, problem: InternalOptimizationProblem, x0: NDArray[np.float64]
+    ) -> InternalOptimizeResult:
+        if not IS_PYSWARMS_INSTALLED:
+            raise NotInstalledError(PYSWARMS_NOT_INSTALLED_ERROR)
+
+        import pyswarms as ps
+
+        base_options = _pso_options_to_dict(self.options)
+        topology_options = {
+            "k": self.topology.k_neighbors,
+            "p": self.topology.p_norm,
+        }
+        pso_options_dict = {**base_options, **topology_options}
+
+        optimizer_kwargs = {
+            "options": pso_options_dict,
+            "static": self.topology.static,
+        }
+
+        res = _pyswarms_internal(
+            problem=problem,
+            x0=x0,
+            optimizer_class=ps.single.LocalBestPSO,
+            optimizer_kwargs=optimizer_kwargs,
+            n_particles=self.n_particles,
```

Comment:
It might also make sense to pass a CommonOptions class (i.e. self) to `_pyswarms_internal` to avoid this long list of arguments. Would be similar to [here](https://github.com/optimagic-dev/optimagic/blob/381285002cff8edfb6c981f95a1b780fc9f19b64/src/optimagic/optimizers/gfo_optimizers.py#L692)